include/entities/entity_factory.hpp

```diff
@@ -14,10 +14,11 @@ class EntityFactory {
 public:
     EntityFactory(std::shared_ptr<EntityManager> em, RenderableFactory& rf);
     ~EntityFactory() = default;
-    int createPanda(double x, double y, int playerId) const;
-    int createGorilla(double x, double y, int playerId) const;
-    int createImage(std::string path, int x, int y, int width, int height, Layers layer, int alpha);
-    int createPlatform(double x, double y, double xScale, double yScale, std::string path, int alpha);
+    int createPanda(double xPos, double yPos, double xScale, double yScale, int playerId) const;
+    int createGorilla(double xPos, double yPos, double xScale, double yScale, int playerId) const;
+    int createWeapon(double xPos, double yPos, double xScale, double yScale) const;
```

Comment:
createWeapon needs to be able to support other weapons in the future. Change method to createPistol or make use of a parameter

Reply:
PJ & me were discussing about refactoring the whole entity_factory, a lot of things could be improved but that might be better in a different pull request

Reply:
I don't think it should be decided how to create the different weapons right now, this is just a test weapon for the PickupSystem. When the actual weapons will get created and loaded from the json, the _right_ way on how to create those will become apparent.

As for reworking the EntityFactory, the functions at are final like panda and gorilla are good, I think we should fix this factory up whenever we need it.